lib/vmselectapi/server.go

```diff
@@ -918,11 +924,11 @@ func (s *Server) processSeriesSearch(ctx *vmselectRequestCtx) error {
 
 	// Send found blocks to vmselect.
 	blocksRead := 0
+	seriesStats := make(map[string]int)
```

Comment:
This map may significantly increase memory usage at `vmstorage` if the query returns millions of unique series with long metric names and long labels.

Currently the most of memory at `vmstorage` during query is used for keeping in memory the list of `TSID` structure for the found series. The size of `TSID` structure is known beforehand and limited contrary to the size of marshaled metric names.